### PR TITLE
Make Membership engagement banner into an A/B test with Contributions

### PR DESCRIPTION
**Pull request by rtyley**
## What does this change?

50% of the audience will be shown a Contributions message on the engagement banner rather than the Membership message.

A fair bit of refactoring of the existing Membership engagement banner functionality was necessary, to make it fit into the AB testing framework - which in particualar has a synchronous `canRun` method, which is problematic given that the banner is not supposed to display at all if the asynchronous check for an adblocker comes back positive.
## What is the value of this and can you measure success?

We want to get a baseline the conversion rate of the Contributions message ("If you use it, if you like it, why not pay for it? It's only fair.") as used in the engagement banner, and also compare it against the Membership message.

https://trello.com/c/HlD5H68N/325-engagement-banner-a-b-test
## Does this affect other platforms - Amp, Apps, etc?

Nope.
## Screenshots

With the Contributions banner (`#ab-MembershipAndContributionsEngagementBanner=contributions`)
image

With the Membership banner (`#ab-MembershipAndContributionsEngagementBanner=membership`)
image
## Request for comment

@desbo

https://trello.com/c/HlD5H68N/325-engagement-banner-a-b-test

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/common.js [hunk 1]
```diff
@@ -390,7 +390,6 @@ define([
                 ['c-accessibility-prefs', accessibilityPrefs],
                 ['c-pinterest', modules.initPinterest],
                 ['c-save-for-later', modules.saveForLater],
-                ['c-show-membership-messages', modules.showMembershipMessages],
```
Comment (desbo) [inline static/src/javascripts/bootstraps/enhanced/common.js]: You can also remove `modules.showMemberShipMessages` itself from this file

File: static/src/javascripts/projects/common/modules/commercial/membership-messages.js [hunk 2]
```diff
@@ -18,115 +22,166 @@ define([
...
-        AU:   'https://membership.theguardian.com/au/supporter',
-        INT:   'https://membership.theguardian.com/supporter'
-    };
+    return function () {
+        var self = this;
 
-    var defaultData = {
-        arrowWhiteRight: svgs('arrowWhiteRight')
-    };
+        this.id = 'MembershipAndContributionsEngagementBanner';
```
Comment (desbo) [inline static/src/javascripts/projects/common/modules/commercial/membership-messages.js]: this file should go in experiments/tests really

Comment (rtyley) [inline static/src/javascripts/projects/common/modules/commercial/membership-messages.js]: I've rebased and moved the entire `membership-messages.js` file into `experiments/tests`

File: static/src/stylesheets/module/_release-message.scss [hunk 3]
```diff
@@ -473,9 +473,9 @@ $btn-height: gs-height(1) - $gs-baseline/2; //30px
 .site-message--membership-message-deep-purple {
     background: #7d0068;
 }
-.site-message--membership-message-orange {
+.site-message--membership-message-yellow {
```
Comment (rtyley) [inline static/src/stylesheets/module/_release-message.scss]: The orange used as one of the Membership message colours is actually used by Contributions, so Ben Wuersching asked that we replace the membership colour with yellow and use the orange (#ffbb00) for the Contributions banner.

**Comment by desbo** [comment]
👍

**Comment by rtyley** [comment]
Verified visible on Production:

https://www.theguardian.com/technology/shortcuts/2016/apr/20/how-vote-leave-got-rickrolled#ab-MembershipAndContributionsEngagementBanner=contributions
image

**Pull request merged**